### ISSUE TITLE
Blackscreen after Nintendo Switch Logo on Mariko & Aula (OLED)

### ISSUE THREAD

**Issue by muxi1** [cathery/sys-ftpd#23]
The above problem occurs on both Switch revisions (Switch Lite & OLED). At first I assumed it was due to the SD card (SanDisk), but it is the same under a Samsung card. 

My system:
AMS 1.4.0 with emuMMC on FW 15.0.0
previously AMS 1.3.2 with emuMMC on FW 14.1.2

**Comment by monstrosities** [comment]
It works fine on an OLED here.

**Comment by muxi1** [comment]
Apart from me, this problem affects several other users that I know of. Interestingly, it does not seem to exist in general. But it only seems to affect consoles that have a modchip installed, even among Erista consoles (with Trinket M0 or SX Core).

**Comment by cathery** [comment]
Do you have any other sysmodules running on the system? If yes, does the system boot if you disable all sysmodules except sys-ftpd?

**Comment by monstrosities** [comment]
> But it only seems to affect consoles that have a modchip installed, even among Erista consoles (with Trinket M0 or SX Core).

I have a modchip of course, how else could I run atmosphere on a Switch OLED? Did this problem start on the latest firmware or also on 14.1.2?

**Comment by muxi1** [comment]
The problem already existed under FW 14.1.2 and 14.1.1, i.e. since I owned an Aula (OLED) Switch. Interestingly, the problem has only existed under my Lite since FW 14.1.2 and currently under FW 15.0.0. However, I don't think it's due to the FW version, because about a year ago I was told by a forum member that all his consoles with a modchip (Erista, Mariko & Aula) had this problem. I couldn't understand this at the time because I didn't own a console with a modchip, only the first Erista revision with the RCM bug. Under that, everything ran and still runs smoothly.

**Comment by cathery** [comment]
Do you have your SD card formatted as exFAT? It's known to cause random corruptions over time, so if you do, can you format it to FAT32 and check once more?

**Comment by muxi1** [comment]
No, I have been using the FAT32 file system since the beginning, as it is generally known that exFAT is unsuitable and leads to data corruption.

**Comment by muxi1** [comment]
I think I have found the cause of the problem!!

 It is Atmosphére's dns_mitm! I use a special CFW package that I put together for my forum, which among other things works with dns_mitm set up to protect CFW users from being banned. I have removed the "default.txt" from atmosphere/hosts and now the system boots with activated sysFTP-light. However, I don't want to do without this protection under any circumstances and would rather use the FTP app instead, should no solution be found here.

**Comment by cathery** [comment]
I was initally going to suggest that your system may simply not have enough memory to run the entirety of sys-ftpd (a classic blackscreen on boot problem is running too many sysmodules at once), but I assumed your system was clean and found it hard to believe it wouldn't have enough memory to run it. Having a mitm service may have changed things, but seeing as it doesn't help brings me back to my confusion

**Comment by monstrosities** [comment]
The fact that you had it working briefly by playing around with dns.mitm makes it seem like it's some kind of network setting, network issue or a problem with your cfw package and not a problem with sys-ftpd. It works for others.

**Comment by muxi1** [comment]
It is not a problem with my CFW package, which is used under all possible switch revisions, by numerous users. As I already said before, everything runs flawlessly under Erista v1 (without modchip), with regard to sysFTPD-Light. The problem apparently does not occur with all switch revisions with modchip, but only in isolated cases. That's all I can say about it. It is always difficult to identify a problem when it occurs randomly. Incidentally, this issue report seems to be about the same problem:
https://github.com/cathery/sys-ftpd/issues/22

**Comment by btc08gh** [comment]
I was getting black screens as well after the 15.0.0 update.  No other sysmodules (completely brand new SDCard).
I have a Trinket M0 installed but I can’t imagine how that would interfere.
I do use dns:mitm.
Removing it fixed my boot issue but I really depend on sys-ftpd for adhoc data transfers.  :-(

**Comment by btc08gh** [comment]
Additional Data Points:

My 14.1.2 Mariko/HWFLY machine continues to have no issues running sys-ftpd.

My 14.1.2 Erista/Trinket M0 machine was fine until I upgraded to 15.0.0.  I'm able to run ftpd pro 3.1.0 with out any issues.

Just not able to have sys-ftpd running as a service even if I delete dns:mitm a/h/default.txt file.

**Comment by btc08gh** [comment]
So…I formatted a new SDCard with nothing but Atmos 15.0.0, Hekate 5.9.0 and sys-ftpd-light…and everything works.  At this point I’m pretty convinced there is something on my old card that’s somehow interfering with sys-ftpd-light so it’s something on my end.  Sincere apologies for any concern caused.  Will see if I can figure it out and possibly help others in the same situation.

**Comment by torbjornaxelsson** [comment]
> So…I formatted a new SDCard with nothing but Atmos 15.0.0, Hekate 5.9.0 and sys-ftpd-light…and everything works. At this point I’m pretty convinced there is something on my old card that’s somehow interfering with sys-ftpd-light so it’s something on my end. Sincere apologies for any concern caused. Will see if I can figure it out and possibly help others in the same situation.

Just installed latest sys-ftpd-light and getting the blank screen.. regular switch V1 with Atmos 15. Did you figure out why? Maybe I'll also try with a new SD card.

**Comment by impeeza** [comment]
guys this is not a support site, please go to discord or GBATemp.

Saying that you could test with a new SD put Hekate y Atmosphère and SYS-FtpD and test you will see everything goes fine.

so you need to check your card for corruption you also can use H2TestW to be sure you card is fine.

**Comment by Lamcza** [comment]
Same issues here first on my Kingstone 128gb then on my Samsung 512 gb. I can boot it from rmc loader once but when want to reboot it stucks on a black screen after tendo logo. Cards are perfectly fine and the problem disappears instantly when sys-ftpd-light is removed from sd.

**Comment by Conan179** [comment]
I can't confirm that, I have an erista with fw 15.0.1, atmos v1.4.0, with sys-ftpd-light and control it via the tesla menu. It always boots, always shuts down properly and I have no problems with restarts.

**Comment by Lamcza** [comment]
> I can't confirm that, I have an erista with fw 15.0.1, atmos v1.4.0, with sys-ftpd-light and control it via the tesla menu. It always boots, always shuts down properly and I have no problems with restarts.

What hetkate version and what other hb/overlays do you use?

**Comment by Conan179** [comment]
v6.0.1 with Nyx v1.5.1
It doesn't matter at this point, the fact is that sys-ftpd-light doesn't cause the switch to have problems booting/restarting/shutting down.

**Comment by Lamcza** [comment]
yes probably not by itself but it is causing black screens on reboots for sure.
